include/zephyr/pm/policy.h

```diff
@@ -158,6 +165,43 @@ void pm_policy_latency_changed_subscribe(struct pm_policy_latency_subscription *
  */
 void pm_policy_latency_changed_unsubscribe(struct pm_policy_latency_subscription *req);
 
+/**
+ * @brief Register an event.
+ *
+ * Events in the power-management policy context are defined as any source that
+ * will wake up the system at a known time in the future. By registering such
+ * event, the policy manager will be able to decide wether certain power states
+ * are worth entering or not.
+ *
+ * @note Expired events need to be unregistered manually using
```

Comment:
It would be nice to be able to remove this responsibility from the API user. But it can be enhanced later if we find it beneficial

Reply:
It's not easy because the PM subsystem doesn't have any link with such events. It just knows _when_ it will happen.

Reply:
What happens if they are not removed ? They eventually be accounted again and in a not intuitive way since it will depend on if a newer event is added or not.

How this is supposed to be used ? My main concern is how it is expected to the user know when to remove it since there is not feedback about when it expired, they should set a timeout ? if so the timeout itself would be accounted by the policy.

What about you register three events `a (now() + 100)`,  `b (now() + 150)`,  `c(now() + 175)`;
It is expected that the policy understands that there will be events within 100 us then another one 50us later and after that one 25us later, but if the no event is removed / added or updated, `next_event_cyc` will account only for `a` . Am I missing something ?